tests/storage/linstor/test_linstor_diskless_resources.py

```diff
@@ -0,0 +1,74 @@
+import pytest
+
+# Requirements:
+# From --hosts parameter:
+# - host(A1): first XCP-ng host >= 8.2 with an additional unused disk for the SR.
+# From --vm parameter
+# - A VM to import to the LINSTOR SR
+# And:
+# - access to XCP-ng RPM repository from hostA1
+
+class TestLinstorDisklessResource:
+    @pytest.mark.small_vm
+    def test_diskless_kept(self, host, linstor_sr, vm_on_linstor_sr):
+        vm = vm_on_linstor_sr
+        vdi_uuids = vm.vdi_uuids(sr_uuid=linstor_sr.uuid)
+        vdi_uuid = vdi_uuids[0]
+        assert vdi_uuid is not None
+
+        controller_option = "--controllers="
+        for h in host.pool.hosts:
+            controller_option += f"{h.hostname_or_ip},"
+
+        # Get volume name from VDI uuid
+        # "xcp/volume/{vdi_uuid}/volume-name": "{volume_name}"
+        output = host.ssh([
+            "linstor-kv-tool", "--dump-volumes", "-g", "xcp-sr-linstor_group_thin_device",
+            "|", "grep", "volume-name", "|", "grep", vdi_uuid
+        ])
+        volume_name = output.split(': ')[1].split('"')[1]
+
+        # Find host where volume is diskless
```

Comment:
We can have many diskless, so >= 1 host(s). We must probably use an array of diskless.